File: 06 JavaScript/controlFlow.js
/* EXPRESIONES CONDICIONALES
Conjunto de instrucciones que se ejecutan si se cumple una condición

--- if
--- else
--- else if
--- switch

==== Declaración if
Sintaxis

    if(condition){
        //bloque de código que se ejecuta si la condición es verdadera
    }
*/
//Ejemplo para validar la mayoría de edad
let edadDaniel = 7;

if(edadDaniel >= 18){
    console.log("Es mayor de edad");
}

/* ==== Declaración else
Se utiliza como instrucción para especificar un bloque de código que se ejecuta si la condición es falsa.
Sintaxis

    if(condition){
        //bloque de código que se ejecuta si la condición es verdadera
    } else {
        //bloque de código que se ejecuta si la condición es falsa
    }
*/

if(edadDaniel >= 18){
    console.log("Es mayor de edad");
} else {
    console.log("Es menor de edad");
}

//Ejemplo de comparación entre dos números con declaración if-else
let num1 = 845369;
let num2 = 4587333;

if(num1 > num2){
    console.log(`El número ${num1} es mayor que ${num2}`);
} else{
    console.log(`El número ${num2} es mayor que ${num1}`);
}

//Segunda forma
let resultado;

if(num1 > num2){
    resultado = ("El número " + num1 + " es mayor que " + num2);
} else{
    resultado = ("El número " + num2 + " es mayor que " + num1);
}

console.log(resultado);

//Determinar si un número es par o impar
let numerito = 18979;

if(numerito % 2 === 0){
    console.log("El número " + numerito + " es par");
}else {
    console.log("El número " + numerito + " es impar");
}

/* ===== Operador ternario =====
Son una forma concisa de escribir una expresión condicional en un lenguaje de programación. Se puede utilizar para realizar la misma tarea de una declaración if-else completa.

Sintaxis
(condicion) ? expresion_true : expresion_false;

*/
//Determinar si un número es par o impar utilizando OPERADOR TERNARIO
let numerito2 = 18978;

(numerito2 % 2 === 0) ? console.log("El número " + numerito2 + " es par") : console.log("El número " + numerito2 + " es impar");

/*
//Tercera forma
let resultadoOperadorTernario = (numerito % 2 === 0) ? ("El número " + numerito + " es par") : ("El número " + numerito + " es impar");

console.log(resultadoOperadorTernario);*/

/* ==== Sentencia else if ====
Se utiliza para especificar una nueva condición si la primera condición es falsa (no se cumple).

Sintaxis
    if(condición1){
        //Si se cumple, se ejecuta este bloque
    }else if(condición2){
        //Si se cumple, se ejecuta este bloque
    }else if(condición3){
        //Si se cumple, se ejecuta este bloque
    }else{
        //Se ejecuta si no se cumplió ninguna de las condiciones anteriores
    }
*/

/*Determinar qué día de la semana corresponde al número, según la siguiente tabla:
    - 1 = Lunes
    - 2 = Martes
    - 3 = Miércoles
    - 4 = Jueves
    - 5 = Viernes
    - 6 = Sábado
    - 7 = Domingo
*/
let dia = 3;

if(dia === 1){
    console.log("Lunes");
}else if(dia === 2){
    console.log("Martes");
}else if(dia === 3){
    console.log("Miércoles");
}else if(dia === 4){
    console.log("Jueves");
}else if(dia === 5){
    console.log("Viernes");
}else if(dia === 6){
    console.log("Sábado");
}else if(dia === 7){
    console.log("Domingo");
}else{
    console.log("Número inválido, intente nuevamente");
}

/* Instrucciones: Crea un programa en JavaScript en donde le muestres al usuario si es de mañana, de tarde, de noche o de madrugada.
Requerimientos:
    - La hora almacenada en una variable (let hora)
    - Establecer el formato de 24 horas (0 - 23 horas)
    - Tablita para determinar los horarios correspondientes a mañana, tarde, noche y madrugada:
        * 6 - 12 es de mañana (hora >= 6 && hora <= 12)
        * 13 - 18 es de tarde (hora >= 13 && hora <=18)
        * 19 - 23 de noche (hora >= 19 && hora <= 23)
        * 0 - 5 es de madrugada (hora >= 0 && hora <= 5)
        * Si hora está fuera de los valores 0 a 23, es inválido
 */

let hora = 24;

if(hora >= 6 && hora <= 12){
    console.log("Es de mañana");
}else if(hora >= 13 && hora <=18){
    console.log("Es de tarde");
}else if(hora >= 19 && hora <= 23){
    console.log("Es de noche, ya que inicia a las 19 horas según la mayoría de la CH28");
}else if(hora >= 0 && hora <= 5){
    console.log("Es de madrugada, aunque nadie da las buenas madrugadas");
}else{
    console.log("Escribe una hora válida entre las 0 y 23");
}

// ==== Sentencias anidadas ====
//Instrucciones: crear un programa para determinar si es mayor de edad o menor de edad. Si es mayor de edad y tiene menos de 65 años, catalogarlo como adulto, si tiene más de 65 años catalogarlo como adulto mayor.
let edadGiovani = 1;

if(edadGiovani >= 18){
    if(edadGiovani < 65){
        console.log("Es un adulto");
    } else {
        console.log("Es un adulto mayor");
    }
}else if(edadGiovani >= 12){
    console.log("Es un adolescente");
}else if(edadGiovani >= 3){
    console.log("Es un niño");
}else{
    console.log("Es un bebé");
}

//Verificar la hora del día mediante una función
const today = new Date(); //Nuevo objeto con un constructor Date(). Sirve para traer la fecha y hora actual.
let hour = today.getHours(); //método para invocar la hora del día en entero con formato de 24 horas
//today.toLocaleTimeString para hora con minutos y segundos

function greeting(){
    if(hour >= 6 && hour <= 12){
        console.log("Es de mañana");
    }else if(hour >= 13 && hour <=18){
        console.log("Es de tarde");
    }else if(hour >= 19 && hour <= 23){
        console.log("Es de noche, ya que inicia a las 19 hours según la mayoría de la CH28");
    }else if(hour >= 0 && hour <= 5){
        console.log("Es de madrugada, aunque nadie da las buenas madrugadas");
    }else{
        console.log("Escribe una hora válida entre las 0 y 23");
    }
}
greeting();

//Verificar la hora del día mediante una función pero de otra manera, guardando los bloques de ejecución en una nueva variable
function saludo(){
    let mensaje = "Error";

    if(hour >= 6 && hour <= 12){
        mensaje = "Buenos días";
    }else if(hour >= 13 && hour <=18){
        mensaje = "Buenas tardes";
    }else if(hour >= 19 && hour <= 23){
        mensaje = "Buenas noches";
    }else{
        mensaje = "Buenas madrugada";
    }
    return mensaje;
}
console.log(saludo());

//Crear un programa en JavaScript que evalúe si un día corresponde a día de la semana (L-V) o fin de semana (S y D).

const checarDia = (diaDeSemana) =>{
    let finDeSemana = diaDeSemana === "Sábado" || diaDeSemana === "Domingo";

    if(finDeSemana){
        return "Es fin de semana";
    }else{
        return "Es día de semana (Lunes a viernes)"
    }
};

console.log(checarDia("Domingo"));

/* ==== SWITCH ====
Permite evaluar la expresión haciendo coincidir el valor de la expresión con una etiqueta case. Si la encuentra, ejecuta la declaración.

Sintaxis

switch(expresion){
    case etiqueta1:
        declaración1;
        break;
    case etiqueta2:
        declaración2;
        break;
    case etiqueta3:
        declaración3;
        break;
    case etiquetaN...........
        declaraciónN;
        break;
    default:
        declaraciónDefault;
        break;
}

*/

/*Determinar mediante un programa que utilice sentencia swtich, qué día de la semana corresponde al número, según la siguiente tabla:
    - 1 = Lunes
    - 2 = Martes
    - 3 = Miércoles
    - 4 = Jueves
    - 5 = Viernes
    - 6 = Sábado
    - 7 = Domingo
*/

let diaSemana = 3;

switch(diaSemana){
    case 1:
        console.log("Lunes");
        break;
    case 2:
        console.log("Martes");
        break;
    case 3:
        console.log("Miércoles");
        break;
    case 4:
        console.log("Jueves");
        break;
    case 5:
        console.log("Viernes");
        break;
    case 6:
        console.log("Sábado");
        break;
    case 7:
        console.log("Domingo");
        break;
    default:
        console.log("Número de la semana inválido");
        break;
}

//Crear un programa con un switch que muestre el precio del kilo de naranjas, manzanas, mandarinas, platanos, mangos

let fruta = "ManDAriNas";

switch(fruta.toLowerCase()){
    case "naranjas":
        console.log("Las naranjas cuestan $30 el kilo");
        break;
    case "manzanas":
        console.log("Las manzanas cuestan $60 el kilo");
        break;
    case "mandarinas":
        console.log("Las mandarinas cuestan $25 el kilo");
        break;
    case "platano":
        console.log("Los platanos cuestan 2 kilos por $25");
        break;
    case "mangos":
        console.log("Los mangos cuestan $22 el kilo");
        break;
    default:
        console.log("No tenemos esa fruta :C");
        break;
}

//Crear un programa con switch para evaluación de alumnos (las calificaciones aprobatorias van de 5.5 a 10 y deben redondearse al número superior inmediato).

File: JAVA_05_controlFlow/ControlFlow.java
package org.generation.controlFlow;

import java.util.Scanner;

public class ControlFlow {

	public static void main(String[] args) {
		/*
		 * if (expresion_logica){
		 * }do while( i = 0; i ++)
		if(expresion_logica){
			//ejecuta un bloque de código
		}else if(expresion_logica){
			//ejecuta este bloque de código
		}else{
			//ejecuta este bloque de código
		}
		*/

		//Ejemplo 1
		int time = 22;
		
		if(time >= 6 && time <=12) {
			System.out.println("Es de mañana");
		}else if(time >= 13 && time <=18) {
			System.out.println("Es de tarde");
		}else if(time >= 19 && time <= 23) {
			System.out.println("Es de noche");
		}else if(time >= 0 && time <= 5) {
			System.out.println("Es de madrugada");
		}else {
			System.out.println("Escribe una hora válida");
		}
		
		//Ejemplo 2. Días de la semana: 1. Lunes --- 7. Domingo
		int day = 9;
		
		if(day == 1) {
			System.out.println("Lunes");
		}else if(day == 2) {
			System.out.println("Martes");
		}else if(day == 3) {
			System.out.println("Miércoles");
		}else if(day == 4) {
			System.out.println("Jueves");
		}else if(day == 5) {
			System.out.println("Viernes");
		}else if(day == 6) {
			System.out.println("Sábado");
		}else if(day == 7) {
			System.out.println("Domingo");
		}else {
			System.out.println("Número de la semana inválido");
		}
		
		/*Java User Input (Scanner)
		 * La clase Scanner se utiliza para obtener la entrada del usuario y se encuentra en el java.util.package.
		 * import java.util.Scanner; De esta manera importamos nuestra clase Scanner desde la JAVA API
		 * */
		
		Scanner scanner = new Scanner(System.in);//Objeto Scanner creado
		System.out.println("Escribe tu nombre de usuario");
		
		String userName = scanner.nextLine();
		
		System.out.println("Escribe tu edad");
		int age = scanner.nextInt();
		
		System.out.println("Tu nombre de usuario es " + userName + " y tu edad es de " + age + " años");
		
		System.out.println("Seleccione un número correspondiente al día de la semana, donde 1 es Lunes hasta 7 es Domingo");
		
		int dia = scanner.nextInt();
		
		if(dia == 1) {
			System.out.println("Lunes");
		}else if(dia == 2) {
			System.out.println("Martes");
		}else if(dia == 3) {
			System.out.println("Miércoles");
		}else if(dia == 4) {
			System.out.println("Jueves");
		}else if(dia == 5) {
			System.out.println("Viernes");
		}else if(dia == 6) {
			System.out.println("Sábado");
		}else if(dia == 7) {
			System.out.println("Domingo");
		}else {
			System.out.println("Número de la semana inválido");
		}
		
		//Switch
		System.out.println("Seleccione un número correspondiente al día de la semana laboral, donde 1 es Lunes hasta 5 es viernes.");
		int diaSemana = scanner.nextInt();
		
		switch(diaSemana) {
		case 1:
			System.out.println("Lunes");
			break;
		case 2:
			System.out.println("Martes");
			break;
		case 3:
			System.out.println("Miércoles");
			break;
		case 4:
			System.out.println("Jueves");
			break;
		case 5:
			System.out.println("Viernes");
			default:
				System.out.println("Número inválido");
		}
		
		scanner.close();		
	}

}
